client/client.go

```diff
@@ -0,0 +1,135 @@
+package client
+
+import (
+	"net/http"
+	"encoding/json"
+	"github.com/pkg/errors"
+	"bytes"
+	"strings"
+	"fmt"
+	"time"
+	"io/ioutil"
+	"strconv"
+	"github.com/microdevs/missy/log"
+)
+// todo: implement token auth
+// MiSSy Client object. Target is the service that will be called
+type Client struct {
+	Target string
+	host string
+	port string
+}
+
+// Create a new Client for a target service
+func New(target string) *Client {
+	//todo: implement service discovery and set host and port
+	return &Client{Target: target}
+}
+
+// Generall Call function can handle all http methods, query string and json marshalling for posts
+func (c *Client) Call(method string, path string, query map[string]string, v interface{}) (*http.Response, error) {
+	//  create a reader in case we need to post or put
+	var dataReader *bytes.Reader
+
+	if method == "POST" || method == "PUT" {
```

Comment:
we should use http.MethodPost and http.MethodPut instead hardcoded strings

Reply:
Yes!